cityofaustin/atd-airflow
# New version of slack library & enhanced slack error message

## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/19797 for the new slack library and https://github.com/cityofaustin/atd-data-tech/issues/20070 for the improved notifications.

I kinda got on a roll and thought I'd play around with the notifications while I was in there, so that's what's up with this unexpected addition to this PR. I think it's worth it though. Here's an example:

![image](https://github.com/user-attachments/assets/171409e2-c127-4075-930a-fbc54a281e25)


LMK if I need to split up this PR into two -- thank you!

## Testing

* Switch your local stack to local build mode and build
* Spin it all up
* Make sure that you have a slack integration in `Admin` => `Connections`, it's in 1pw
* Fire off the slack notification tester DAG
* Edit the slack notification test DAG itself and play with the `byline` key to edit the DAG error byline
* 👀 Try out the optional `icon` key for instant DAG visibility
* Observe the way identities are encoded for mentions
* Observe how it decodes cron schedules
  * `Every 15 minutes, between 09:00 AM and 04:59 PM, Monday through Friday, March through September` is rendered from `*/15 9-16 * 3-9 1-5` for example
* Consider how exception classes could be used to better categorize error modes


---
#### Ship list
